util/json.go

```diff
@@ -21,7 +22,20 @@ import (
 func UnmarshalJSON(bs []byte, x interface{}) (err error) {
 	buf := bytes.NewBuffer(bs)
 	decoder := NewJSONDecoder(buf)
-	return decoder.Decode(x)
+	if err := decoder.Decode(x); err != nil {
+		return err
+	}
+
+	// Since decoder.Decode validates only the first json structure in bytes,
+	// check if decoder has more bytes to consume to validate whole input bytes.
+	tok, err := decoder.Token()
+	if tok != nil {
+		return fmt.Errorf("invalid JSON input caused by '%s', expected EOF", tok)
```

Comment:
Can we change the error message to be more similar to what the standard `json.Unmarshal` gives?
It has an error message like: `error: invalid character '{' after top-level value`

Ex:  https://play.golang.org/p/dN0ttkqe1O2